packages/@aws-cdk/aws-cloudwatch/test/test.metrics.ts

```diff
@@ -122,6 +133,45 @@ export = {
       } );
     }, /The maximum number of dimensions is 10, received 11/);
 
+    test.throws(() => {
+      new Metric({
+        namespace: 'Test',
+        metricName: 'ACount',
+        period: cdk.Duration.minutes(10),
+        dimensionsMap: {
+          dimensionA: 'value1',
+          dimensionB: 'value2',
+          dimensionC: 'value3',
+          dimensionD: 'value4',
+          dimensionE: 'value5',
+          dimensionF: 'value6',
+          dimensionG: 'value7',
+          dimensionH: 'value8',
+          dimensionI: 'value9',
+          dimensionJ: 'value10',
+          dimensionK: 'value11',
+        },
+      } );
+    }, /The maximum number of dimensions is 10, received 11/);
+
+    test.done();
+  },
+
+  'cannot use both dimensions and dimensionsMap'(test: Test) {
+    test.throws(() => {
+      new Metric({
+        namespace: 'Test',
+        metricName: 'ACount',
+        period: cdk.Duration.minutes(10),
+        dimensions: {
+          dimensionA: 'value1',
+        },
+        dimensionsMap: {
+          dimensionA: 'value1',
+        },
+      });
+    }, /Using both the 'dimensions' and 'dimensionsMap' properties is not supported. Use only 'dimensionsMap.'/);
+
     test.done();
   },
 };
```

Comment:
I feel like we're missing some positive test case where only the new `dimensionsMap` property was passed...?